herbert_core/classes/MPIFramework/get_num_cores.m

```diff
@@ -0,0 +1,16 @@
+function [physical, logical] = get_num_cores()
+% Get the number of physical and logical cores on the system
+%
+% Output:
+% -------
+%   physical      The number of physical cores on this PC
+%   logical       The number of logical cores on this PC
+%
+core_info = evalc('feature(''numcores'')');
+match_str = 'MATLAB detected: ([0-9]+) %s cores';
+
+[match, ~] = regexp(core_info, sprintf(match_str, 'physical'), 'tokens', 'match');
```

Comment:
should this line processing be another helper function? you do the same line processing twice:
```
core_info = evalc('feature(''numcores'')');
physical = get_element(core_info, 'physical')
logical = get_element(core_info, 'logical')
```

Reply:
if you're feeling enthusiastic the `get_element()` method can then be tested with some hard-coded test strings to verify it's pulling the correct data.